File: isabelle.eclipse.core/src/isabelle/eclipse/core/util/SafeSessionActor.java
package isabelle.eclipse.core.util;

import isabelle.eclipse.core.IsabelleCorePlugin;
import isabelle.scala.SessionActor;

/**
 * A session actor that handles all exceptions by logging them to plug-in log
 * 
 * @author Andrius Velykis
 */
public class SafeSessionActor extends SessionActor {

	@Override
	public void handleException(Exception e) {
		// log exception (parent rethrows it)
		IsabelleCorePlugin.log(e);
	}

}

File: isabelle.eclipse.core/src/isabelle/eclipse/core/util/SafeSessionActor.scala
package isabelle.eclipse.core.util

import isabelle.scala.SessionActor
import isabelle.eclipse.core.internal.IsabelleCorePlugin.{error, log}


/**
 * A session actor that handles all exceptions by logging them to plug-in log
 *
 * @author Andrius Velykis
 */
class SafeSessionActor extends SessionActor {

  // log exception (parent rethrows it)
  override def handleException(e: Exception) = log(error(Some(e)))

}
